group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroup.java

```diff
@@ -538,17 +537,46 @@ public void validateOffsetCommit(
         if (memberEpoch < 0 && members().isEmpty()) return;
 
         final ConsumerGroupMember member = getOrMaybeCreateMember(memberId, false);
-        if (memberEpoch != member.memberEpoch()) {
-            throw Errors.STALE_MEMBER_EPOCH.exception();
-        }
+        validateMemberEpoch(memberEpoch, member.memberEpoch());
     }
 
     /**
      * Validates the OffsetFetch request.
+     *
+     * @param memberId              The member id for consumer groups.
+     * @param memberEpoch           The member epoch for consumer groups.
+     * @param lastCommittedOffset   The last committed offsets in the timeline.
      */
     @Override
-    public void validateOffsetFetch() {
-        // Nothing.
+    public void validateOffsetFetch(
+        String memberId,
+        int memberEpoch,
+        long lastCommittedOffset
+    ) throws UnknownMemberIdException, StaleMemberEpochException {
+        // When the member epoch is -1, the request comes from the admin client. In this case,
+        // the request can commit offsets if the group is empty.
```

Comment:
Should we check the group is empty here? The code seems misaligned with the comment.

Reply:
I think the "when the member epoch is -1" part applies here. Is the empty group part somewhere else?

Reply:
I made a comment on something similar in another PR.  Might be a slightly different case, but similar wording about committing offsets is allowed and it being a bit unclear if fetching them is or isn't

https://github.com/apache/kafka/pull/14120/files/7e5f1f64d173d3e3b68fd6310fb76d6137daed2c#r1303310001

Reply:
The comment is wrong. Let me fix it.